### PR TITLE
[ND-20] Create Signup Form

### PR DESCRIPTION
**Pull request by DrMalina**
- Added SignUp Form

- Prefixed Login Components with `Styled` (EDIT: Reverted that part )

- Updated Yup's validation schemes

- Changed name from `Props` to `FormikProps` (that way it is more semantic)

- Created helper.js file to store commonly used functions

https://node-wro.atlassian.net/browse/ND-20

### CODE REVIEW

File: frontend/src/pages/SignIn/SignIn.js [hunk 1]
```diff
@@ -19,11 +16,11 @@ const SignIn = () => (
...
     }}
     validationSchema={Yup.object().shape({
-      username: userNameValidation,
-      password: passwordValidation
+      username: fieldValidation('Username is required'),
+      password: fieldValidation('Password is required')
     })}
   >
-    {props => {
+    {formik => {
```
Comment (Reactor94) [inline frontend/src/pages/SignIn/SignIn.js]: I think that for someone this name **formik** could be a little bit confused, so i would name it **formikProps**

File: frontend/src/components/Button/Button.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import React from 'react';
 
-const Button = ({ type, disabled, children, ...rest }) => {
+const StyledButton = ({ type, disabled, children, ...rest }) => {
   return (
     <button type={type} disabled={disabled} {...rest}>
       {children}
     </button>
   );
 };
```
Comment (dommie73) [inline frontend/src/components/Button/Button.js]: In my task [ND-26] currently I have something like this:

- `Button.styles.js`:
``` javascript
import styled from 'styled-components';
const Button = styled.button` (...) `;
export { Button };
```
- `Button.js`:
``` javascript
import React from 'react';
import * as S from './Button.styles';
const Button = ({type, disabled, children, ...rest}) => {
  return (
    <S.Button type={type} disabled={disabled} {...rest}>
      {children}
    </S.Button>
  );
}
export { Button };
```
- `index.js`
``` javascript
export { Button } from './Button';
```
What do you think about it? @DrMalina AFAIR this is the file structure you've proposed on Slack.

Comment (DrMalina) [reply frontend/src/components/Button/Button.js]: Right, let stick to what you have, it makes more sense.

File: frontend/src/pages/SignIn/SignIn.js (hunk 1 above)
Comment (DrMalina) [reply frontend/src/pages/SignIn/SignIn.js]: Good idea!

**Pull request merged**